File: reverse_integer/ReverseInteger.py
class Solution(object):
    def reverse(self, x):
        is_neg = False
        if x < 0:
            is_neg = True 
            x = -x
        result = 0
        while x != 0:
            result = 10*result + x%10
            x = int( x / 10 )
        if result > 2147483647: return 0
        if is_neg:
            result = result * -1
        return result

#test
if __name__ == "__main__":
    sol = Solution()
    print sol.reverse(-123)

File: cpp/reverse_integer.cpp
class Solution {
public:
    int reverse(int x) {
        bool isNegative = ( x < 0);
        x = abs(x);
        int result = 0;
        while (x > 0){
            if (result > ((INT_MAX - (x%10)) / 10)) return 0;
            result *= 10;
            result += x%10;
            x = x / 10;
        };
        return (isNegative ? (-1)*result : result );
    }
};

